File: java/src/Ice/ObjectWriter.java
// **********************************************************************
//
// Copyright (c) 2003-2009 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************

package Ice;

public abstract class ObjectWriter extends ObjectImpl
{
    public abstract void write(OutputStream out);

    public void
    __write(IceInternal.BasicStream os)
    {
        OutputStream stream = (OutputStream)os.closure();
        write(stream);
    }

    public void
    __read(IceInternal.BasicStream is, boolean rid)
    {
        assert(false);
    }
}

File: cs/src/Ice/ObjectWriter.cs
// **********************************************************************
//
// Copyright (c) 2003-2007 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************

using System.Diagnostics;

namespace Ice {

    public abstract class ObjectWriter : ObjectImpl
    {
        public abstract void write(OutputStream outStream);

        public override void write__(IceInternal.BasicStream os)
        {
            IceInternal.BasicOutputStream bos = (IceInternal.BasicOutputStream)os;
            write(bos.out_);
        }

        public override void read__(IceInternal.BasicStream istr, bool rid)
        {
            Debug.Assert(false);
        }
    }

}
